src/main/java/eu/bbmri_eric/negotiator/database/model/Request.java

```diff
@@ -52,7 +52,7 @@ public class Request {
   @Column(name = "id")
   private String id;
 
-  @ManyToMany
+  @ManyToMany(fetch = FetchType.EAGER)
```

Comment:
this will produce a large overhead. Can this be lazy?

Reply:
This is linked to the similar comment above and this change is due to the fact then, when the getCurrentState per resource is called by the resource lifecycle service, the negotiation object is not "complete", because it does not contain the child reference to the request (and then to the resource). So the getResources() method will throw a LAZY exception. That was one of the reasons why I had thought about implementing and calling the repository for the NegotiationResourceLifecycleRecord directly from the recource lifecucle service, moving out from the Negotiation class this logic, since now the state is not get anymore from a Negotiation's internal key-value table. Fore sure we can inspect (if you have suggestion) a way to avoid both the EAGERS and the repository implementation

Reply:
please remove it a use something like a named entity graph, or a custom method in the repository. We cannot have Eager annotations on every property.

Reply:
Fixed in 18f3761c23e6d941a2cd8c7053b98734ce3cd6a8